app/models/katello/repository.rb

```diff
@@ -72,15 +72,14 @@ class Repository < Katello::Model
       :with => /^([a-z0-9\-_]{4,30}\/)?[a-z0-9\-_\.]{3,30}$/,
       :message => (_("must be a valid docker name"))
     }
+    validates :url, :allow_nil => :custom?, :if => :in_default_view?,
+      :url_schema => ['http', 'https', 'ftp', 'file']
```

Comment:
Haven't check yet, but we need to make sure that you can use URL's with basic auth after this change. i.e.:

http://user:password@foo.example.com/bar